### PR TITLE
Add support for MS SQL Server

### PR DESCRIPTION
**Pull request by tsdogs**
This adds support for MS SQL Server to migrations of this extension.
There are modifications to the base Migration class, and a few fixes to migrations for handling correctly up/down actions.
SQL Server by default sets NOT NULL to fields so I added explicitly ' NULL' to the statements where required.

Have tested with SQL Server 2012 and mysql/mariadb 
not tested with postgresql but it should be basically the same.

### CODE REVIEW

**Comment by tsdogs** [comment]
Actually I just discovered that the NULL or NOT NULL default on not specified columns in SQL Server can be set from the database options or with the following:
SET ANSI_NULL_DFLT_OFF ON;

though I think that specifying it will avoid confusion anyway.
I though had to enable this option in the database as the migration for rbac would work, but set all fields as NOT NULL by default.

**Comment by tsdogs** [comment]
I also added the handling of tablePrefix in the names of the indexes (as it could create conflicts in some rare cases when the same database is used by more applications with tablePrefix enabled)

File: migrations/Migration.php [hunk 1]
```diff
@@ -33,12 +37,39 @@ public function init()
         switch (Yii::$app->db->driverName) {
             case 'mysql':
                 $this->tableOptions = 'CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE=InnoDB';
+                $this->dbType = 'mysql';
                 break;
             case 'pgsql':
                 $this->tableOptions = null;
+                $this->dbType = 'mysql';
```
Comment (dmeroff) [inline migrations/Migration.php]: This should be `pgsql`

**Comment by dmeroff** [comment]
I'm sorry for the delay with reviewing your PR.
I've added comment on your code, could you please fix it, and I'll merge PR.

**Comment by tsdogs** [comment]
Should be ok now.

File: yii2-user.kdev4 [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[Project]
```
Comment (dmeroff) [inline yii2-user.kdev4]: This file should be removed.

**Comment by dmeroff** [comment]
Thank you! Great job!

**Comment by tsdogs** [comment]
Great, glad you intend to continue developing this.

**Pull request merged**